samples/employee-service/src/main/java/io/spring/sample/graphql/SalaryService.java

```diff
@@ -0,0 +1,23 @@
+package io.spring.sample.graphql;
+
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+import java.math.BigDecimal;
+
+@Component
+public class SalaryService {
+
+
+    @PreAuthorize("hasRole('MANAGER')")
+    public Mono<BigDecimal> getSalaryForEmployee(Employee employee) {
```

Comment:
Mixing Servlet and reactive has always been a fragile situation when involving contexts. The problem is that in a Servlet environment the `SecurityContext` is saved in a `ThreadLocal` which won't work when using Reactive because there is no guarantee of what Thread the context will operate on.